tests/unit/test_uploader.py

```diff
@@ -128,13 +128,13 @@ def test_upload_hypervisor_details_to_scc_retry(self, collected_results):
                                         url=scc_url))
 
         uploader = SCCUploader(scc_creds)
-
-        with mock.patch('requests.put', return_value=FakeResponse(429)
+        with mock.patch('requests.put', return_value=FakeResponse(429, {'Retry-After': 5})
```

Comment:
Probably could reduce this to 1 second delay, so as to not increase unit test run time unnecessarily...

Note also that time.sleep can take a floating point seconds value to sleep for, so you could actually switch to the retry_upload() taking a float or int as possible values, and use 0.001 here, and get the same effect in terms of log message getting generated, without having to wait for any real time having passed ;-)

Reply:
Set the delay to 1 sec